widget_periodictable/periodic_table.py

```diff
@@ -21,29 +22,78 @@ class PTableWidget(DOMWidget):
     _view_module = Unicode(module_name).tag(sync=True)
     _view_module_version = Unicode(module_version).tag(sync=True)
-    selected_elements = List([]).tag(sync=True)
+    selected_elements = Dict({}).tag(sync=True)
     disabled_elements = List([]).tag(sync=True)
     display_names_replacements = Dict({}).tag(sync=True)
     disabled_color = Unicode('gray').tag(sync=True)
-    noselect_color = Unicode('pink').tag(sync=True)
+    unselected_color = Unicode('pink').tag(sync=True)
     states = Int(1).tag(sync=True)
-    selected_states = List([]).tag(sync=True)
     selected_colors = List([]).tag(sync=True)
+    allElements = List([
+        "H", "He", "Li", "Be", "B", "C", "N", "O", "F", "Ne", "Na", "Mg",
+        "Al", "Si", "P", "S", "Cl", "Ar", "K", "Ca", "Sc", "Ti", "V", "Cr", "Mn", "Fe",
+        "Co","Ni", "Cu", "Zn", "Ga", "Ge", "As", "Se", "Br", "Kr", "Rb", "Sr", "Y",
+        "Zr", "Nb", "Mo",  "Tc", "Ru", "Rh","Pd", "Ag", "Cd", "In", "Sn", "Sb", "Te",
+        "I", "Xe", "Cs", "Ba", "Hf", "Ta", "W", "Re", "Os", "Ir","Pt", "Au", "Hg", "Tl",
+        "Pb", "Bi", "Po", "At", "Rn", "Fr", "Ra", "Rf", "Db", "Sg", "Bh", "Hs",
+        "Mt","Ds", "Rg", "Cn", "Nh", "Fi", "Mc", "Lv", "Ts", "Og", "La", "Ce", "Pr",
+        "Nd", "Pm", "Sm", "Eu","Gd",  "Tb", "Dy", "Ho", "Er", "Tm", "Yb", "Lu","Ac",
+        "Th", "Pa", "U", "Np", "Pu", "Am","Cm", "Bk",  "Cf", "Es", "Fm", "Md", "No", "Lr"
+    ]).tag(sync=True)
 
-
-    def __init__(self, states = 1, disabled_color = 'gray', noselect_color = 'pink', selected_colors = ["#a6cee3", "#b2df8a", "#fdbf6f", "#6a3d9a", "#b15928", "#e31a1c", "#1f78b4", "#33a02c", "#ff7f00", "#cab2d6", "#ffff99"]):
+    def __init__(self, states = 1, selected_elements = {}, disabled_elements = [], disabled_color = 'gray', unselected_color = 'pink', selected_colors = ["#a6cee3", "#b2df8a", "#fdbf6f", "#6a3d9a", "#b15928", "#e31a1c", "#1f78b4", "#33a02c", "#ff7f00", "#cab2d6", "#ffff99"]):
         super(PTableWidget, self).__init__()
         self.states = states
         self.disabled_color = disabled_color
-        self.noselect_color = noselect_color
+        self.disabled_elements = disabled_elements
+        self.unselected_color = unselected_color
         self.selected_colors = selected_colors
+        self.selected_elements = selected_elements
 
         if len(selected_colors) < states:
-            self.selected_colors = selected_colors + ["#a6cee3", "#b2df8a", "#fdbf6f", "#6a3d9a", "#b15928", "#e31a1c", "#1f78b4", "#33a02c", "#ff7f00", "#cab2d6", "#ffff99"]
+            additional_colors = ["#a6cee3", "#b2df8a", "#fdbf6f", "#6a3d9a", "#b15928", "#e31a1c", "#1f78b4", "#33a02c", "#ff7f00", "#cab2d6", "#ffff99"]
+            self.selected_colors = selected_colors + additional_colors * (1 + (states - len(selected_colors)) // len(additional_colors))
+            self.selected_colors = self.selected_colors[:states]
+
+
+    def set_element_state(self, elementName, state):
```

Comment:
I don't think this function is ever used. Hence the functionality of the widget is non-existing at the moment.

Reply:
This function is import to set the element states.

For example:
```python
PTable.selected_elements["H"] = 2
```
will not work. 

On can use this function to set individual element's state.